recipe/meta.yaml

```diff
@@ -34,32 +33,38 @@ requirements:
     - rich
     - namex
     - h5py
-    - dm-tree
+    - optree
     - ml_dtypes
-    # this is not mentioned in the upstream until version 3.3.0
-    # but it might have been a bug, because it is indeed used
-    # in this version as well at:
-    # https://github.com/keras-team/keras/blob/v3.0.5/keras/utils/torch_utils.py#L3
     - packaging
   run_constrained:
-    # as suggested in https://keras.io/getting_started/#tensorflow-compatibility
-    # tensorflow pinning expands ~=2.16.1
-    - tensorflow >=2.16.1,<2.17.0a
-    # should be 0.4.20, but this will allow the jax[cuda] version to be installed (0.4.24)
-    - jax >=0.4.20, <0.5.0a
-    # pytorch should have >=2.1.0,<2.2.0 (expanding ~=2.1.0) but in 
-    # https://github.com/keras-team/keras/blob/v3.0.5/requirements.txt#L6
-    # https://github.com/keras-team/keras/blob/v3.0.5/keras/utils/torch_utils.py#L156
-    # a simple >=2.1.0 lower bound is used.
+    # https://github.com/keras-team/keras/blob/v{{ version }}/requirements.txt
+    - tensorflow >=2.17.0,<3.0a
+    # should be 0.4.23, but this will allow the jax[cuda] version to be installed (0.4.24)
+    - jax >=0.4.23,<0.5.0a
```

Comment:
This one is very unclear. I don't remember why we used `>=0.4.20` in the previous PR and their docs are contradicting themselves... Anyway, I think it's fine. You might want to remove the comment above.